lib/logstash/inputs/s3.rb

```diff
@@ -109,6 +112,12 @@ def register
     if !@watch_for_new_files && original_params.include?('interval')
       logger.warn("`watch_for_new_files` has been disabled; `interval` directive will be ignored.")
     end
+
+    @gzip_detector = case @decode_gzip
+                     when "force"  then -> (_) { true }
+                     when "detect" then -> (filename) { filename.end_with?('.gz','.gzip') }
+                     else fail(LogStah::ConfigurationError, "unsupported value `#{@decode_gzip}` for decode_gzip")
```

Comment:
`Logstah::ConfigurationError` is missing an 'S'. Otherwise, LGTM pending successful CI.

Reply:
🤦‍♂